BinInfoUtils.h

```diff
@@ -12,25 +12,37 @@ typedef std::pair<int, int>                 BinInfo;
 typedef std::vector<std::vector<BinInfo>>   BinInfoLayerMap;
 typedef std::vector<BinInfoLayerMap>        BinInfoMap;
 
+#ifdef __CUDACC__
+__host__ __device__
```

Comment:
We could have a derived define for those ... so we don't need to have ifdefs all over the place.